src/Storages/NATS/NATSHandler.cpp

```diff
@@ -27,6 +27,13 @@ NATSHandler::NATSHandler(LoggerPtr log_)
     : log(log_)
     , loop_state(Loop::CREATED)
 {
+    auto error = uv_async_init(loop.getLoop(), &execute_tasks_scheduler, processTasks);
+    if (error)
+    {
+        throw Exception(ErrorCodes::CANNOT_CONNECT_NATS, "Can not create scheduler");
```

Comment:
Maybe also add the received error in some way to the error message, so we can have at least some idea why it has failed.
```suggestion
        throw Exception(ErrorCodes::CANNOT_CONNECT_NATS, "Cannot create scheduler, received error {}", error);
```